tests/test_client_functional.py

```diff
@@ -2167,6 +2167,29 @@ def dumps(obj):
             assert False, "never executed"  # pragma: no cover
 
 
+async def test_raise_for_status_coro(aiohttp_client) -> None:
```

Comment:
Let's have same method for the custom callable `raise_for_status` passed per request, similar to test pair `test_raise_for_status` and `test_raise_for_status_per_request`

Reply:
Please read https://github.com/aio-libs/aiohttp/issues/3864#issuecomment-505152107

> `response_check` should definitely be added to `ClientSession`, if I want custom logic on a request I can do it after response as usual.

Reply:
well, since `def request(...)` now supports any callable as `raise_for_status`, we would still need a test for it

Reply:
done.